src/render/draw_fill_extrusion.js

```diff
@@ -138,6 +138,13 @@ function flatRoofsUpdate(context, source, coord, bucket, layer, terrain) {
     ];
 
     const getLoadedBucket = (nid) => {
+        if (nid.overscaledZ === nid.canonical.z) {
+            const n = source.getTileByID(nid.key);
+            if (n && n.hasData()) {
+                const nBucket: ?FillExtrusionBucket = (n.getBucket(layer): any);
+                if (nBucket) return nBucket;
+            }
+        }
```

Comment:
Now that we have to repeat this chunk of code, perhaps there's a way to refactor the `getLoadedBucket` function for simplicity, e.g. cutting this bit out as a separate function and getting rid of the loop later?

Reply:
Addressed in a9d4c16a47b90dd464a93c2cefd6c0c4d1bbc2b0.